### ISSUE TITLE
Bug in LARC when using FP16

### ISSUE THREAD

**Issue by a157801** [NVIDIA/apex#1030]
https://github.com/NVIDIA/apex/blob/154c6336aa7aedd40d3b3583fb5e1328f9cdf387/apex/parallel/LARC.py#L101

When using fp16 training, p.grad has been scaled up with the loss scaler,  but the parameter not.  I think the weight-decay also should be scaled up, and the gradient norm should be divided by scale^2
